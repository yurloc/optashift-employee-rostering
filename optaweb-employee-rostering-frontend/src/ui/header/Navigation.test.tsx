/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { NavItem } from '@patternfly/react-core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { getRouterProps } from 'util/BookmarkableTestUtils';
import { Navigation, NavigationProps } from './Navigation';

describe('Navigation', () => {
  it('should activate a navigation link matching the current path', () => {
    const props: NavigationProps = {
      tenantId: 0,
      variant: 'horizontal',
      ...getRouterProps('/0/skills', {}),
    };

    const navigation = shallow(<Navigation {...props} />);
    expect(toJson(navigation)).toMatchSnapshot();

    // NavItem matching the path should be active
    const navItems = navigation.find(NavItem).filterWhere(navItem => navItem.props().itemId === 'skills');
    expect(navItems).toHaveLength(1);
    expect(navItems.at(0).props().isActive).toEqual(true);

    // Other NavItems should be inactive
    navigation.find(NavItem).filterWhere(navItem => navItem.props().itemId !== 'skills').forEach(
      navItem => expect(navItem.props().isActive).toEqual(false),
    );
  });
});
