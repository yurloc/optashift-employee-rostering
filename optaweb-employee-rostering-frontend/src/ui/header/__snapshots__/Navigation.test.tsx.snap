// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should activate a navigation link matching the current path 1`] = `
<Nav
  aria-label="Nav"
  className=""
  onSelect={[Function]}
  onToggle={[Function]}
>
  <NavList
    ariaLeftScroll="Scroll left"
    ariaRightScroll="Scroll right"
    className=""
    variant="horizontal"
  >
    <NavItem
      isActive={true}
      itemId="skills"
      key="skills"
    >
      <Link
        to="/0/skills"
      >
        Trans(i18nKey=skills)
      </Link>
    </NavItem>
    <NavItem
      isActive={false}
      itemId="spots"
      key="spots"
    >
      <Link
        to="/0/spots"
      >
        Trans(i18nKey=spots)
      </Link>
    </NavItem>
    <NavItem
      isActive={false}
      itemId="contracts"
      key="contracts"
    >
      <Link
        to="/0/contracts"
      >
        Trans(i18nKey=contracts)
      </Link>
    </NavItem>
    <NavItem
      isActive={false}
      itemId="employees"
      key="employees"
    >
      <Link
        to="/0/employees"
      >
        Trans(i18nKey=employees)
      </Link>
    </NavItem>
    <NavItem
      isActive={false}
      itemId="shift"
      key="shift"
    >
      <Link
        to="/0/shift"
      >
        Trans(i18nKey=shift)
      </Link>
    </NavItem>
    <NavItem
      isActive={false}
      itemId="availability"
      key="availability"
    >
      <Link
        to="/0/availability"
      >
        Trans(i18nKey=availability)
      </Link>
    </NavItem>
    <NavItem
      isActive={false}
      itemId="rotation"
      key="rotation"
    >
      <Link
        to="/0/rotation"
      >
        Trans(i18nKey=rotation)
      </Link>
    </NavItem>
  </NavList>
</Nav>
`;
